overview/content-types/option.nix

```diff
@@ -0,0 +1,123 @@
+{
+  lib,
+  pkgs,
+  config,
+  ...
+}:
+let
+  inherit (lib)
+    types
+    mkOption
+    join
+    drop
+    optionalString
+    take
+    ;
+in
+{
+  options = {
+    prefix-length = mkOption {
+      type = types.int;
+      default = 2;
+    };
+    loc = mkOption {
+      type = with types; listOf str;
+    };
+    type = mkOption {
+      type = types.str;
+    };
+    default = mkOption {
+      type = types.attrs;
+      default = { };
+    };
+    description = mkOption {
+      type = types.str;
+    };
+    readOnly = mkOption {
+      type = types.bool;
+    };
+    option-prefix = mkOption {
+      type = types.str;
+      readOnly = true;
+      default =
+        let
+          prefix-head = take config.prefix-length config.loc;
+          prefix-tail = drop config.prefix-length config.loc;
+        in
+        ''
+          <span class="option-prefix">${join "." prefix-head}.</span><span>${join "." prefix-tail}</span>
+        '';
+    };
+    option-type = mkOption {
+      type = types.str;
+      readOnly = true;
+      default = ''
+        <dt>Type:</dt>
+        <dd class="option-type"><code>${config.type}</code></dd>
+      '';
+    };
+    option-default = mkOption {
+      type = types.str;
+      readOnly = true;
+      default = optionalString (config.default ? text) ''
+        <dt>Default:</dt>
+        <dd class="option-default"><code>${config.default.text}</code></dd>
+      '';
+    };
+    option-description = mkOption {
+      type = types.str;
+      readOnly = true;
+      default =
+        let
+          # This doesn't actually produce a HTML string but a Jinja2 template string
+          # literal, that is then replaced by it's HTML translation at the last build
+          # step.
+          markdownToHtml = markdown: "{{ markdown_to_html(${builtins.toJSON markdown}) }}";
+        in
+        ''
+          <div class="option-description">
+          ${markdownToHtml config.description}
+          </div>
+        '';
+    };
+    alert-readonly = mkOption {
+      type = types.str;
+      readOnly = true;
+      default = optionalString config.readOnly ''
+        <span class="option-alert" title="This option can't be set by users">Read-only</span>
+      '';
```

Comment:
Should simply be part of rendering the default value,  I see no reason to have a separate field.

Reply:
This makes the `__toString` more readable IMO, but I assume that this comment also applies to the other options as well, right?

Reply:
I see. That makes sense, thanks!